### PR TITLE
Provide read-only support for overridden fork points

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/BaseGitMacheteBranch.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import com.virtuslab.gitmachete.backend.api.IGitMacheteCommit;
 import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus;
 
-@Getter
+@Getter(onMethod_ = {@Override})
```
Comment (PawelLipski) [inline backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/BaseGitMacheteBranch.java:13]: Unrelated change (only thematically connected - all about `overrides` here :bangbang:)

File: gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java [hunk 2]
```diff
@@ -64,7 +65,63 @@ public GitCoreRepository(Path mainDirectoryPath, Path gitDirectoryPath) throws G
...
+
+  private ObjectId existingRevisionToObjectId(String revision) throws GitCoreException {
+    return revisionToObjectId(revision)
+        .getOrElseThrow(() -> new GitCoreNoSuchRevisionException("Commit '${revision}' does not exist in this repository"));
+  }
+
+  private Option<ObjectId> revisionToObjectId(String revision) throws GitCoreException {
+    try {
+      return Option.of(jgitRepo.resolve(revision));
+    } catch (IOException e) {
```
Comment (micpiotrowski) [inline gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java:118]: Vavr.Try ?

File: config/checkstyle/tree_walker_module_directives.xml [hunk 3]
```diff
@@ -168,13 +168,18 @@
...
     <property name="ignoreComments" value="true"/>
     <property name="format" value="\bupdateUI\b"/>
-    <property name="message" value="Probably unintended updateUI() usage. See docs and consider repaint() and revalidate()"/>
+    <property name="message" value="Probably unintended JComponent#updateUI() usage. See docs and consider repaint() and revalidate()"/>
+</module>
+<module name="Regexp">
+    <property name="illegalPattern" value="true"/>
+    <property name="ignoreComments" value="true"/>
+    <property name="format" value="\bclass [A-Za-z0-9]*Util\b"/>
+    <property name="message" value="Use ...Utils (not ...Util) naming convention"/>
```
Comment (mkondratek) [inline config/checkstyle/tree_walker_module_directives.xml:177]: :h*****-kettle:

Comment (PawelLipski) [reply config/checkstyle/tree_walker_module_directives.xml:177]: :triggered: :moral-panic:
